From 6158bce4a9bb5df2796e9b1cba1dd15fbc5e4d24 Mon Sep 17 00:00:00 2001
From: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@intel.com>
Date: Fri, 13 Jun 2014 13:21:40 -0700
Subject: [PATCH 10/10] power_spply: Add null pointer check in prop get/set
 routines

Not all power supply drivers provide property get/set
routines. So dereferncing these function pointers before
checking for NULL value will result in kernel panic. So,
Added a null pointer checks before accesing power supply
property get/set routines.

Change-Id: I4d67db989fed0c3fc9fc58fe48ae4fef45de1d7d
Signed-off-by: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@intel.com>
---
 drivers/power/power_supply_charger.h | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/power/power_supply_charger.h b/drivers/power/power_supply_charger.h
index 0e1d9db..4f64a70 100644
--- a/drivers/power/power_supply_charger.h
+++ b/drivers/power/power_supply_charger.h
@@ -69,9 +69,14 @@ static inline int set_ps_int_property(struct power_supply *psy,
 {
 
 	union power_supply_propval val;
+	int ret = -ENODEV;
 
 	val.intval = prop_val;
-	return psy->set_property(psy, psp, &val);
+
+	if (psy->set_property)
+		ret = psy->set_property(psy, psp, &val);
+
+	return ret;
 }
 
 static inline int get_ps_int_property(struct power_supply *psy,
@@ -81,7 +86,9 @@ static inline int get_ps_int_property(struct power_supply *psy,
 
 	val.intval = 0;
 
-	psy->get_property(psy, psp, &val);
+	if (psy->get_property)
+		psy->get_property(psy, psp, &val);
+
 	return val.intval;
 }
 /* Define a TTL for some properies to optimize the frequency of
-- 
2.0.0

